src/webview/components/memory-table.tsx

```diff
@@ -329,9 +370,88 @@ export class MemoryTable extends React.PureComponent<MemoryTableProps, MemoryTab
             endAddress: startAddress + memoryTableOptions.bigWordsPerRow
         };
     }
+
+    /**
+     * Ensures that the {@link MemoryTableState.groupsPerRowToRender} is correctly set when {@link MemoryTableProps.groupsPerRow} = `Autofit` is used.
+     */
+    protected ensureGroupsPerRowAutofit(): void {
+        if (this.props.groupsPerRow === 'Autofit') {
+            const groupsPerRowToRender = this.determineGroupsPerRowToRender();
+            if (this.state.groupsPerRowToRender !== groupsPerRowToRender) {
+                this.setState(prev => ({ ...prev, groupsPerRowToRender }));
+            }
+        }
+
+    }
+
+    protected determineGroupsPerRowToRender(): number {
+        const options = MemorySizeOptions.create(this.props, this.state);
+
+        if (this.props.groupsPerRow === 'Autofit') {
+            const container = this.datatableRef.current?.getElement().querySelector<HTMLElement>(`td.${DataColumn.CLASS_NAME}`);
+            if (container) {
+                return MemoryTable.approximateGroupsPerRow(container, options);
+            }
+            return 1;
+        }
+
+        return options.groupsPerRow;
+    }
 }
 
 export namespace MemoryTable {
     export const TABLE_CLASS = 'memory-inspector-table';
     export const GROUP_SEPARATOR = 'group-separator';
+
+    /**
+     * The approximation is done by:
+     *
+     * 1. Calculating the number of characters that would fit into the element
+     * 2. Calculating the number of characters that we require
+     * 3. Dividing the available characters by the characters we require
+     *    to aquire the groups per row we theoretically could fit into the element
+     * 4. Including other factors such as margins
+     *
+     * **Note**
+     *
+     * The returned `groups per row` value is based on {@link CONFIG_GROUPS_PER_ROW_NUMERIC_CHOICES}
+     *
+     * @returns Approximated groups per row that will fit into the element
+     */
+    export function approximateGroupsPerRow(element: HTMLElement, options: MemorySizeOptions): number {
```

Comment:
And move this code into the `DataColumn` contribution, behind some acceptably generic column contribution API function.